File: src/main/java/gg/mineral/practice/traits/Spectatable.java
package gg.mineral.practice.traits;

import gg.mineral.practice.entity.Profile;
import gg.mineral.practice.util.collection.ProfileList;
import org.bukkit.World;

import java.util.concurrent.ConcurrentLinkedDeque;

public interface Spectatable {

    ConcurrentLinkedDeque<Profile> getSpectators();

    ProfileList getParticipants();

    boolean isEnded();

    World getWorld();
}

File: src/main/kotlin/gg/mineral/practice/traits/Spectatable.kt
package gg.mineral.practice.traits

import gg.mineral.practice.util.collection.ProfileList
import org.bukkit.World
import java.lang.ref.WeakReference

interface Spectatable {
    val spectators: ProfileList

    val participants: ProfileList

    val ended: Boolean

    val world: WeakReference<World>
}
